types/alpinejs/index.d.ts

```diff
@@ -0,0 +1,225 @@
+
+interface XDataContext {
+    /**
+     * Will be executed before Alpine initializes teh rest of the component.
+     */
+    init?(): void;
+    [stateKey: string]: any;
+}
+type XData = XDataContext | string | number | boolean;
+
+interface DirectiveUtilities {
+    Alpine: Alpine;
+    effect: () => void;
+    cleanup: () => void;
+    evaluate: (expression: string) => unknown;
+    evaluateLater: (expression: string) => (result: unknown) => void;
+}
+interface DirectiveParameters {
+    value: string;
+    modifiers: string[];
+    expression: string;
+    original: string;
+    type: string;
+}
+
+export interface Alpine {
+    // following TSDoc under MIT was taken from
+    // https://github.com/vuejs/vue-next/blob/92f11d6740929f5b591740e30ae5fba50940ec82/packages/reactivity/src/reactive.ts#L65-L87
+    /**
+     * Creates a reactive copy of the original object.
+     * The reactive conversion is "deep"—it affects all nested properties. In the
+     * ES2015 Proxy based implementation, the returned proxy is **not** equal to the
+     * original object. It is recommended to work exclusively with the reactive
+     * proxy and avoid relying on the original object.
+     *
+     * A reactive object also automatically unwraps refs contained in it, so you
+     * don't need to use `.value` when accessing and mutating their value:
+     *
+     * ```js
+     * const count = ref(0)
+     * const obj = reactive({
+     *   count
+     * })
+     *
+     * obj.count++
+     * obj.count // -> 1
+     * count.value // -> 1
+     * ```
+     */
+    reactive<T extends object>(target: T): UnwrapNestedRefs<T>;
+    /**
+     * Releases/stops a reactive effect, if it is currently active.
+     *
+     * @param runner the reactive effect runner to be stopped
+     */
+    release(runner: ReactiveEffectRunner): void;
+    /**
+     * As soon as effect is called, it will run the provided callback
+     * functionbut actively look for any interactions with reactive
+     * data. If it detects an interaction (a get or set from the
+     * aforementioned reactive proxy) it will keep track of it and
+     * make sure to re-run the callback if any of reactive data
+     * changes in the future.
+     *
+     * @param callback the function to be run while looking for reactive interactions
+     */
+    effect(callback: (() => unknown) | ReactiveEffectRunner, options?: ReactiveEffectOptions): ReactiveEffectRunner;
+    /**
+     * Returns the raw, original object of a reactive proxy. This is an escape
+     * hatch that can be used to temporarily read without incurring proxy
+     * access/tracking overhead or write without triggering changes. It is not
+     * recommended to hold a persistent reference to the original object.
+     * Use with caution.
+     *
+     * @param observed the reactive, proxied object
+     */
+    raw<T>(observed: T): T;
```

Comment:
same as `reactive`
https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/alpine.js#L23
```suggestion
    raw: unknown;
```